src/AdvancedApp.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { Props } from 'react';
+import './App.scss';
+
+const goodsFromServer: string[] = [
+  'Dumplings',
+  'Carrot',
+  'Eggs',
+  'Ice cream',
+  'Apple',
+  'Bread',
+  'Fish',
+  'Honey',
+  'Jam',
+  'Garlic',
+];
+
+type State = {
+  goodsInCart: string[];
+};
+
+class App extends React.Component<Props<State>, State> {
```

Comment:
We don't really have any Props here, so you can just use a blank object literal as props type
```suggestion
class App extends React.Component<{}, State> {
```